source/MaterialXGenShader/ShaderGenerator.cpp

```diff
@@ -400,4 +402,84 @@ ShaderNodeImplPtr ShaderGenerator::createCompoundImplementation(const NodeGraph&
     return CompoundNode::create();
 }
 
+void ShaderGenerator::finalizeShaderGraph(ShaderGraph& graph)
+{
+    // Find all thin-film nodes and reconnect them to the 'thinfilm' input
+    // on BSDF nodes layered underneath.
```

Comment:
The thin_film_brdf node is not an ordinary BSDF node. Instead when it's layered on top of another BSDF we reconnect it as an input instead, so the thinfilm parameters are input into the BSDF node. Since the implementation consists of selecting the new Fresnel model (Airy) to use inside the BSDF.

So here we do this re-wiring as custom edits to the shader graph.